src/util/taproot.rs

```diff
@@ -451,6 +451,18 @@ impl TaprootBuilder {
         self.branch.len() == 1 && self.branch[0].is_some()
     }
 
+    /// Checks if the builder has hidden parts.
+    pub fn has_hidden(&self) -> bool {
```

Comment:
nit: 
`has_hidden_branches` or `has_hidden_leaves` instead? I am not a big fan of names where humans have to reason about things in order to complete them "has hidden what?", "Oh, must be `has_hidden_leaves`".

Reply:
Originally I did it as `has_hidden_branches`. Than, I had a though, "may be this is leaves?". Than I have found that the only unambiguous guess present in current API is `add_hidden`, and decided to skip the suffix. No idea how "branch" is different from "node", "part" etc - we have no terminology in place yet in this area.

Reply:
I'm out of my depth here but if 'parts' is correct in the rustdoc then wouldn't `has_hidden_parts` be a descriptive name?

Reply:
"Hidden" can be a script leaf or a tree branch, thus we need some term to describe it. I see only two applicable terms here: `hidden_tips` and `hidden_parts`. But then we also have to rename `TreeBuilder::add_hidden`, and I am not happy with both `add_hidden_tip` and `add_hidden_part`. This also brings us to https://github.com/rust-bitcoin/rust-bitcoin/pull/924#issuecomment-1088232833 where `TipNode` can be used as an enum name for both script leafs and hidden (something).

Reply:
Had come to conclusion that "hidden something" can always be called a "node". Updated all names accordingly.